### PR TITLE
Add gateway monitoring to the probe

### PR DESCRIPTION
**Pull request by geobeau**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ func ParseConfig() Config {
 	config := Config{
 		ConsulAddr: flag.String("consul", "localhost:8500", "Consul server address"),
 		Tag:        flag.String("tag", "s3", "Tag to search on consul"),
-		GatewayTag: flag.String("GatewayTag", "s3", "Tag to search on consul"),
-		EndpointSuffix: flag.String("suffix", ".service.consul.prod.crto.in:30053",
+		GatewayTag: flag.String("GatewayTag", "s3-gateway", "Tag to search on consul"),
+		EndpointSuffix: flag.String("suffix", ".service.{dc}.consul.preprod.crto.in",
```
Comment (erebe) [inline config/config.go]: .preprod ?

File: probe/consul.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+	for i := range destinations {
+
+		endpointEntries, _, err := health.Service(destinations[i].service, "", false, &consul_api.QueryOptions{Datacenter: destinations[i].datacenter})
+		if err != nil {
+			log.Printf("Consul query failed for %s (dc: %s, service: %s): %s", destinations[i].raw, destinations[i].datacenter, destinations[i].service, err)
+		}
+		endpointName := getEndpointFromConsul(destinations[i].service, *cfg.EndpointSuffix, endpointEntries)
+		minioClient, err := newMinioClientFromEndpoint(endpointName, *cfg.AccessKey, *cfg.SecretKey)
+		if err != nil {
```
Comment (erebe) [inline probe/consul.go:79]: Nothing happen if we are in error.
In the next line S3client will be invalid

File: probe/consul.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	raw        string
+}
+
+func extractDestinations(serviceEntries []*consul_api.ServiceEntry) (destinations []destination) {
+	ok := false
+	rawDestinations := ""
+	for i := range serviceEntries {
+		rawDestinations, ok = serviceEntries[i].Service.Meta["gateway_destinations"]
+		if ok {
+			break
```
Comment (erebe) [inline probe/consul.go]: Same here, if we are in error there is no guarantee that rawDestinations contains a meaningful value

File: probe/consul.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	return port
+}
+
+func getDatacenter(serviceEntries []*consul_api.ServiceEntry) string {
+	dc := ""
+	for i := range serviceEntries {
+		dc = serviceEntries[i].Node.Datacenter
+		break
+	}
+	return dc
```
Comment (erebe) [inline probe/consul.go]: should return an error if ""

File: probe/consul_test.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+func TestGenerateEndointFromConsulWithoutProxyData(t *testing.T) {
+	entries := getTestServiceEntries()
+	endpoint := getEndpointFromConsul("test", ".{dc}.prod", entries)
+	if endpoint != "test.us-east-1.prod:8080" {
+		t.Errorf("Failed to generate URL from Consul data")
+	}
+}
+
+func TestGenerateEndointFromConsulWithProxyData(t *testing.T) {
```
Comment (erebe) [inline probe/consul_test.go:38]: Check also some invalid scenarii

**Review by erebe — APPROVED** [review]

**Comment by geobeau** [comment]
Should be ready for another review

**Review by erebe — APPROVED** [review]

**Pull request merged**